### ISSUE TITLE
Slow performance with NetCDF driver

### ISSUE THREAD

**Issue by letmaik** [OSGeo/gdal#620]
## Expected behavior and actual behavior.

NetCDF files can be opened with the NetCDF or the HDF5 driver. When the file has chunked compression enabled with a chunking pattern not matching 1,1,1...,n then performance is extremely slow when invoking any gdal operations.

## Steps to reproduce the problem.

Create test data:
```py
from netCDF4 import Dataset
import numpy as np
rootgrp = Dataset("test.nc", "w", format="NETCDF4")
lat = rootgrp.createDimension("lat", 1800)
lon = rootgrp.createDimension("lon", 3600)
data = rootgrp.createVariable("data", "f4", ('lat', 'lon'), zlib=True, chunksizes=(900, 1800))
data[:] = np.random.rand(1800, 3600)
rootgrp.close()
```

Fast (1 second):
```
gdalinfo -mm -stats test.nc --debug ON
# or:
gdalinfo -mm -stats HDF5:test.nc://data --debug ON
```

Slow (minutes):
```
gdalinfo -mm -stats NETCDF:test.nc:data --debug ON
```

## Operating system

Windows 10 64 bit

## GDAL version and provenance

2.2.2 via conda

**Comment by rouault** [comment]
Known issue: https://trac.osgeo.org/gdal/ticket/5291

**Comment by rouault** [comment]
Likely fixed per a4b05dfe6cc52aa816ec0dafdaac4796c125fd7f
